Let models provider their own specific special tokens

helunwencser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #4255
* #4235
* #4234
* __->__ #4227

Differential Revision: [D59651199](https://our.internmc.facebook.com/intern/diff/D59651199/)

helunwencser committed:
Let models provider their own specific special tokens



Differential Revision: [D59651199](https://our.internmc.facebook.com/intern/diff/D59651199/)

[ghstack-poisoned]

helunwencser committed:
Update on "Let models provider their own specific special tokens"


Differential Revision: [D59651199](https://our.internmc.facebook.com/intern/diff/D59651199/)

[ghstack-poisoned]

helunwencser committed:
Update on "Let models provider their own specific special tokens"


Differential Revision: [D59651199](https://our.internmc.facebook.com/intern/diff/D59651199/)

[ghstack-poisoned]

larryliu0820 reviewed (commented):
I like composition more than inheritance. Do you think it's better to keep the `Tiktoken` class and pass the special tokens through constructor? This way we don't have to create one class per model architecture.

helunwencser committed:
Update on "Let models provider their own specific special tokens"


Differential Revision: [D59651199](https://our.internmc.facebook.com/intern/diff/D59651199/)

[ghstack-poisoned]

helunwencser committed:
Update on "Let models provider their own specific special tokens"


Differential Revision: [D59651199](https://our.internmc.facebook.com/intern/diff/D59651199/)

[ghstack-poisoned]

helunwencser committed:
Update on "Let models provider their own specific special tokens"


Differential Revision: [D59651199](https://our.internmc.facebook.com/intern/diff/D59651199/)

[ghstack-poisoned]
